diff --git a/src/joebkt/ChunkMap.java b/src/joebkt/ChunkMap.java
index e9e2a8e..d681bcd 100644
--- a/src/joebkt/ChunkMap.java
+++ b/src/joebkt/ChunkMap.java
@@ -63,9 +63,7 @@ public class ChunkMap {
         if (this.c.isEmpty()) {
             WorldProvider var5 = this.m_worldServer.worldProvider;
 
-            if (!var5.isOverworld_ContainsSpawn()) {
                 this.m_worldServer.cachedChunks.addChunkXYZsFromLoadedChunkList();
-            }
         }
 
     }
diff --git a/src/joebkt/ServerChunkCache.java b/src/joebkt/ServerChunkCache.java
index 1648828..26e35ee 100644
--- a/src/joebkt/ServerChunkCache.java
+++ b/src/joebkt/ServerChunkCache.java
@@ -55,18 +55,9 @@ public class ServerChunkCache implements IChunkProvider {
     }
 
     public void markChunkEligibleForUnload(int var1, int var2) {
-        if (this.m_worldServer.worldProvider.isOverworld_ContainsSpawn()) {
-            if (!this.m_worldServer.isInside128OfSpawnMaybe(var1, var2)) {
-                this.setOfChunksToSaveAndUnload.add(
-                        Long.valueOf(
-                                ChunkCoordIntPair.combineTwoIntsIntoLong(var1,
-                                var2)));
-            }
-        } else {
             this.setOfChunksToSaveAndUnload.add(
                     Long.valueOf(
                             ChunkCoordIntPair.combineTwoIntsIntoLong(var1, var2)));
-        }
 
     }
 
diff --git a/src/net/minecraft/server/MinecraftServer.java b/src/net/minecraft/server/MinecraftServer.java
index b678e81..f32e368 100644
--- a/src/net/minecraft/server/MinecraftServer.java
+++ b/src/net/minecraft/server/MinecraftServer.java
@@ -370,6 +370,7 @@ public abstract class MinecraftServer implements CommandSender, Runnable, vn, wd
     }
 
     protected void generateTerrain(int argWorldServerIdx) {
+       return;/*
         boolean var1 = true;
         boolean var2 = true;
         boolean var3 = true;
@@ -400,6 +401,7 @@ public abstract class MinecraftServer implements CommandSender, Runnable, vn, wd
         }
 
         this.clearGeneratingSpawnNotice();
+        */
     }
 
     protected void setResourcePackInResourcesZip(String var1, IDataManager var2) {
