diff --git a/WrapperObjects/ItemStackWrapper.java b/WrapperObjects/ItemStackWrapper.java
index b1a3fdd..2f0f776 100644
--- a/WrapperObjects/ItemStackWrapper.java
+++ b/WrapperObjects/ItemStackWrapper.java
@@ -126,7 +126,6 @@ public class ItemStackWrapper implements MC_ItemStack {
             }
 
             EnchantmentManager.applyEnchantsToItem(map, this.is);
-            System.out.println("Set " + map.size() + "enchants.");
         }
     }
 
